akka-persistence-testkit/src/main/scala/akka/persistence/testkit/internal/TestKitStorage.scala

```diff
@@ -35,6 +37,8 @@ sealed trait InMemStorage[K, R] extends InternalReprSupport[R] {
 
   import scala.math._
 
+  final val eventsLog: ConcurrentHashMap[K, util.Queue[InternalRepr]] =
+    new ConcurrentHashMap()
   private final val eventsMap: ConcurrentHashMap[K, (Long, Vector[InternalRepr])] =
```

Comment:
By just looking at the code and the names here in InMemStorage I think it's very unclear what eventsMap and eventsLog are used for. Do they need to be in sync, for example should `clearAll` also clear eventsLog? Do they need to be updated atomically for concurrent operations?

Reply:
I can only come up with a comment explaining the semantics of both. Would that be sufficient?

Reply:
Is the `eventsLog` only used via `expectNextPersisted`?
If that is the case, could we name it to indicate that as the purpose?
`expectNextQueue`?

`first` and `removeFirst` are also too vague. `firstInExpectNextQueue` and `removeFirstInExpectNextQueue`?

Reply:
Sounds good 👍